src/hooks/usePlaceOrder.ts

```diff
@@ -0,0 +1,76 @@
+import BN from 'bn.js'
+import { toast } from 'react-toastify'
+
+import { TokenDetails, PlaceOrderParams as ExchangeApiPlaceOrderParams } from 'types'
+import { useState } from 'react'
+import { exchangeApi } from 'api'
+import { log } from 'utils'
+import { txOptionalParams } from 'utils/transaction'
+import { useWalletConnection } from './useWalletConnection'
+import { DEFAULT_ORDER_DURATION } from 'const'
+
+interface Params {
+  callback: () => void
+}
+
+interface PlaceOrderParams {
+  buyAmount: BN
+  buyToken: TokenDetails
+  sellAmount: BN
+  sellToken: TokenDetails
+}
+
+interface Result {
+  placeOrder: (params: PlaceOrderParams) => Promise<void>
+  isSubmitting: boolean
+}
+
+export const usePlaceOrder = ({ callback }: Params): Result => {
```

Comment:
Why are you using callback instead of handling the promise returned form `placeOrder`? A promise is more versatile, imho

Reply:
Not sure that's what you had in mind, but I did refactor out the `callback`. Please take a look.

Reply:
You did exactly what I meant :+1: